meddlr/metrics/lpips_meddlr.py

```diff
@@ -0,0 +1,163 @@
+        mode: str = "grayscale",
+        lpips: bool = True,
+        pretrained: bool = True,
+        channel_names: Sequence[str] = None,
+        reduction="none",
+        compute_on_step: bool = False,
+        dist_sync_on_step: bool = False,
+        process_group: bool = None,
+        dist_sync_fn: bool = None,
+    ):
+        """
+        Args:
+        net_type (str): The pre-trained network to use for extracting features. One of:
+            * ``'alex'``: Alex-Net w/ feature extraction layers 'relu1' through 'relu5'
+            * ``'vgg'``: VGG-16 w/ feature extration layers ['relu1_2', 'relu2_2',
+                       'relu3_3', 'relu4_3', 'relu5_3']
+            * ``'squeeze'``: Squeeze-Net w/ feature extration layers 'relu1' through 'relu7'
+        mode (str): Determines how to interpret the channel dimension of the inputs. One of:
+            * ``'grayscale'``: Each channel corresponds to a distinct grayscale input image.
+            * ``'rgb'``: The 3 channel dimensions correspond to a single rgb image.
+                         Exception will be thrown if channel dimension != 3 dtype data is complex.
+        lpips (bool): This flag determines if a linear layer is used on top of the
+                      extracted features.
+            * ``True``: linear layers on top of base/trunk network.
+            * ``False``: no linear layers; each layer is averaged together.
+        pretrained (bool): This flag controls the linear layers, which are only in
+                           effect when lpips=True above.
+            * ``True``: linear layers are calibrated with human perceptual judgments.
+            * ``False``: linear layers are randomly initialized.
+        """
+
+        super().__init__(
+            channel_names=channel_names,
+            units="",
+            reduction=reduction,
+            compute_on_step=compute_on_step,
+            dist_sync_on_step=dist_sync_on_step,
+            process_group=process_group,
+            dist_sync_fn=dist_sync_fn,
+        )
+
+        if not env.package_available("lpips"):
+            raise ModuleNotFoundError(
+                "LPIPS metric requires that lpips is installed."
+                "Either install as `pip install meddlr[metrics]` or `pip install lpips`."
+            )
+
+        valid_net_type = ("vgg", "alex", "squeeze")
+        if net_type not in valid_net_type:
+            raise ValueError(
+                f"Invalid `net_type` ('{net_type}'). Expected one of {valid_net_type}."
+            )
+
+        valid_modes = ("grayscale", "rgb")
+        if mode not in valid_modes:
+            raise ValueError(f"Invalid `mode` ('{mode}'). Expected one of {valid_modes}.")
+
+        self.net = NoTrainLpips(net=net_type, lpips=lpips, verbose=False)
+        self.mode = mode
+
+    def func(self, preds: torch.Tensor, targets: torch.Tensor):
+
+        if self.mode == "grayscale":
+            loss_shape = (targets.shape[0], targets.shape[1])
+
```

Comment:
nit: remove empty line in between `if/elif` clauses. this occurs at other locations as well, check those